app/src/main/java/com/github/se/travelpouch/model/activity/map/DirectionsRepository.kt

```diff
@@ -29,12 +32,83 @@ class DirectionsRepository(private val client: OkHttpClient) : DirectionsReposit
 
     val routesArray = jsonObject.getJSONArray("routes")
 
+    // Parse routes using Kotlin's functional style
     val routes =
         List(routesArray.length()) { i ->
           val routeObject = routesArray.getJSONObject(i)
+
+          // Extract overview polyline for the route
           val overviewPolylineObject = routeObject.getJSONObject("overview_polyline")
           val points = overviewPolylineObject.getString("points")
-          Route(OverviewPolyline(points))
+
+          // Extract the legs array using optJSONArray to avoid JSONException
+          val legsArray = routeObject.optJSONArray("legs")
+
+          // Check if legsArray is null or empty
+          val legs =
+              if (legsArray != null && legsArray.length() > 0) {
```

Comment:
You can simplify it as :
```suggestion
         val legs = if (legsArray?.length() ?: 0 > 0) {
```

Reply:
It's a good idea but the problem is that legsArray remains nullable in the `if` condition if we don't do the explicit check and the compiler is complaining about it. It is therefore simpler to keep this version